src/Instrumentation/ReactHTTP/src/ReactHttpInstrumentation.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\ReactHttp;
+
+use OpenTelemetry\API\Globals;
+use OpenTelemetry\API\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use function OpenTelemetry\Instrumentation\hook;
+use OpenTelemetry\SemConv\TraceAttributes;
+use Psr\Http\Message\ResponseInterface;
+use React\Http\Io\Transaction;
+use React\Http\Message\ResponseException;
+use React\Promise\PromiseInterface;
+use Throwable;
+
+/** @psalm-suppress UnusedClass */
+class ReactHttpInstrumentation
+{
+    public const NAME = 'react-http';
+
+    public static function register(): void
+    {
+        $instrumentation = new CachedInstrumentation(
+            'io.opentelemetry.contrib.php.react-http',
+            schemaUrl: 'https://opentelemetry.io/schemas/1.32.0',
+        );
+
+        /** @psalm-suppress UnusedFunctionCall */
+        hook(
+            Transaction::class,
+            'send',
+            pre: static function (Transaction $transaction, array $params, string $class, string $function, ?string $filename, ?int $lineno) use ($instrumentation): ?array {
+                $request = $params[0];
+
+                $propagator = Globals::propagator();
+                $parentContext = Context::getCurrent();
+
+                foreach ($propagator->fields() as $field) {
+                    $request = $request->withoutHeader($field);
+                }
+
+                /** @psalm-suppress ArgumentTypeCoercion */
+                $spanBuilder = $instrumentation
+                    ->tracer()
+                    // https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-client-span
+                    ->spanBuilder(sprintf('%s', $request->getMethod()))
+                    ->setParent($parentContext)
+                    ->setSpanKind(SpanKind::KIND_CLIENT)
+                    ->setAttribute(TraceAttributes::HTTP_REQUEST_METHOD, $request->getMethod())
+                    ->setAttribute(TraceAttributes::SERVER_ADDRESS, $request->getUri()->getHost())
+                    ->setAttribute(TraceAttributes::SERVER_PORT, $request->getUri()->getPort())
+                    ->setAttribute(TraceAttributes::URL_FULL, (string) $request->getUri())
+                    ->setAttribute(TraceAttributes::NETWORK_PROTOCOL_NAME, 'http')
+                    ->setAttribute(TraceAttributes::NETWORK_PROTOCOL_VERSION, $request->getProtocolVersion())
+                    ->setAttribute(TraceAttributes::HTTP_REQUEST_BODY_SIZE, $request->getHeaderLine('Content-Length'))
+                    ->setAttribute(TraceAttributes::URL_SCHEME, $request->getUri()->getScheme())
+                    ->setAttribute(TraceAttributes::USER_AGENT_ORIGINAL, $request->getHeaderLine('User-Agent'))
```

Comment:
These attributes are marked as opt-in:
> Instrumentations SHOULD populate the attribute if and only if the user configures the instrumentation to do so.

Reply:
Thanks! I didn't have my browser wide enough to see the requirement level column in the table. Fixing in the next commit.